### DISCUSSION TITLE
Плавность Эффектов и Бегущей строки.

### DISCUSSION THREAD

**Discussion by shkolin1973** [vvip-68/GyverPanelWiFi#316] [Общие вопросы]
Здравствуйте!
Не могу добиться плавности эффектов на новых версиях ядра и FastLed на ESP8266   (WiFiPanel v.1.13.2024.1106).
Версия ядра ESP8266 - 3.1.2   
Версия FastLED      - 3.6.0; 3.7.0. Так же проверил: 3.7.6; 3.7.7; 3.7.8
С эффектами ладно, не так заметно. Я всегда тестирую на эффекте "Матрица". Там сразу видны срывы.
Вот с бегущей строкой хуже, там сразу заметно подергивание. Иногда чуть замедление и опять ускорение. Нет плавности.
Решил перепрошить на новую версию только из за погоды от Яндекса.
На ESP32:
Версия ядра ESP32   - 2.0.14
Версия FastLED      - 3.6.0
Новая версия работает идеально.

В итоге вернулся на предыдущую версию на  ESP8266:
Версия ядра ESP8266 - 2.7.4   
Версия FastLED      - 3.5.0
По мне это идеальное сочетание. На эффекте "Матрица" ни одного срыва. И бегущая строка плавная.

**Comment by vvip-68** [comment]
Да, для ESP8266 ядро 2.7.4 самое беспроблемное. Я ядрами более поздних версий у FastLED заметные проблемы, выражающиеся либо в отсутствии плавности, либо вообще не некоторых эффектах/режимах мерцает только самый первый светодиод.

В гитхабе FastLED куча issues по этому поводу. Проблема известная.
На последней версии FastLED 3.9.4 контроллеры на 8266 вообще к сети подключиться не могут...

Если повезет - можно перебирать комбинации параметров компиляции - с каким-то сочетанием возможно не будет ни зависания ни подергиваний. Как повезет. В данном случае для версии 1.13 - самое правильное и беспроблемное решение - оставаться на версии с ядром 2.7.4

**Reply by shkolin1973** [reply]
"оставаться на версии с ядром 2.7.4"- так похоже и придётся делать для ESP8266. Плохо, что последняя версия уже не компилируется на:
Версия ядра ESP8266 - 2.7.4
Версия FastLED - 3.5.0

**Reply by vvip-68** [reply]
Уже компилируется. Поправил / выложил час назад...

**Reply by shkolin1973** [reply]
Получилось! 
Версия ядра ESP8266 - 2.7.4
Версия FastLED - 3.5.0
Скомпилировалось, записалось, работает. Пока на гирлянде смотанной в рулон. Вечером проверю на развернутой матрице на плавность.
Вы еще и новые эффекты добавили!!!
Спасибо большое!

**Reply by shkolin1973** [reply]
Здравствуйте! Проверил новую (WiFiPanel v.1.13.2024.1212) версию прошивки.
Хороший результат на:
На ESP32:
Версия ядра ESP32 - 2.0.17 (и 2.0.14)
Версия FastLED - 3.6.0

На ESP8266:
Версия ядра ESP8266 - 2.7.4
Версия FastLED - 3.5.0

Спасибо!
